applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -315,28 +396,53 @@ static void connected_cb(struct bt_conn *conn, uint8_t err)
 		LOG_ERR("Failed to set security to L2: %d", ret);
 	}
 
-	/* TODO: Channel nr is 0 for now, to be changed with dual CIS */
-	ble_vcs_discover(conn, 0);
+	if (!ble_acl_gateway_all_links_connected()) {
+		ble_acl_start_scan();
+	} else {
+		bt_le_scan_stop();
+	}
 }
 
 static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
 {
 	char addr[BT_ADDR_LE_STR_LEN];
 
-	if (conn != default_conn) {
+	if (conn != headset_conn[AUDIO_CHANNEL_LEFT] && conn != headset_conn[AUDIO_CHANNEL_RIGHT]) {
 		return;
 	}
 
 	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
 	LOG_INF("Disconnected: %s (reason 0x%02x)", addr, reason);
 
-	bt_conn_unref(default_conn);
-	default_conn = NULL;
+	if (conn == headset_conn[AUDIO_CHANNEL_LEFT]) {
+		bt_conn_unref(headset_conn[AUDIO_CHANNEL_LEFT]);
```

Comment:
Since there is no bt_conn_ref in this file, do we need to unref?

Reply:
Maybe we need a bt_conn_ref in the device_found, when we set headset_conn[index] = conn?